sdk/servicebus/Azure.Messaging.ServiceBus/api/Azure.Messaging.ServiceBus.netstandard2.0.cs

```diff
@@ -777,6 +780,31 @@ public sealed partial class TrueRuleFilter : Azure.Messaging.ServiceBus.Manageme
         public override string ToString() { throw null; }
     }
 }
+namespace Azure.Messaging.ServiceBus.Plugins
+{
+    public abstract partial class ServiceBusPlugin
+    {
+        protected ServiceBusPlugin() { }
+        public abstract string Name { get; }
+        public virtual bool ShouldContinueOnException { get { throw null; } }
+        public virtual System.Threading.Tasks.Task AfterMessageReceive(Azure.Messaging.ServiceBus.ServiceBusReceivedMessage message) { throw null; }
+        public virtual System.Threading.Tasks.Task BeforeMessageSend(Azure.Messaging.ServiceBus.ServiceBusMessage message) { throw null; }
+        protected static void SetBody(Azure.Messaging.ServiceBus.ServiceBusReceivedMessage message, Azure.Core.BinaryData body) { }
```

Comment:
Yes; the terminology and pattern are used across the messaging space for track one.   Service Bus, in particular, has an active third party ecosystem of plugins today.

Reply:
The Set methods are necessary because the `ServiceBusReceivedMessage` is read-only. If someone wants to override this by opting into plugins, we provide an avenue for doing so that doesn't impact the mainline scenarios.

Reply:
Has Krzysztof seen this yet?  This feels a little _too_ different to me without understanding more of the tradeoffs that led to this design.  We should schedule an hour next week to chat about it.

Reply:
(And don't block the PR on that discussion)

Reply:
Yeah we will definitely need to review with Krzysztof. We have a bunch of API changes/additions that are coming in for Preview 4. I will schedule something for next week.

Reply:
As an external collaborator, I have no idea what's the focus of the attention and would love to understand. I can't expect you, to plan on inviting me for your internal call, but if it works out, I'd be happy to join.

Reply:
I think the focus will be on Track 2 patterns for plugins, as well as some of the concerns around the Set* methods for setting properties on a received message. It would be great if you can attend! 
/cc @danielmarbach in case you are interested in the topic of plugins and would like to attend as well.